src/iothread.c

```diff
@@ -617,6 +780,13 @@ int processClientsFromMainThread(IOThread *t) {
             connSetReadHandler(c->conn, readQueryFromClient);
         }
 
+        /* Cache a pointer to the master so we can quickly check if it needs
+         * to be send to main thread for its replication cron in case it's
+         * waiting for long inside the IO thread */
+        if (c->flags & CLIENT_MASTER) {
+            t->master = c;
+        }
```

Comment:
maybe we simplify it
```suggestion
        /* Cache the master client. */
        if (c->flags & CLIENT_MASTER) t->master = c;
```